sympy/polys/matrices/domainscalar.py

```diff
@@ -0,0 +1,83 @@
+"""
+
+Module for the DomainScalar class.
+
+A DomainScalar represents an element which is in a particular
+Domain. The idea is that the DomainScalar class provides the
+convenience routines for unifying elements with different domains.
+
+It assists in Scalar Multiplication and getitem for DomainMatrix.
+
+"""
+from ..constructor import construct_domain
+
+
+class DomainScalar:
+    r"""
+    docstring
+    """
+
+    def __init__(self, element, domain):
+        self.element = element
+        self.domain = domain
+
+    @classmethod
+    def new(cls, element, domain):
+        return cls(element, domain)
+
+    def __repr__(self):
+        return repr(self.element)
+
+    @classmethod
+    def from_sympy(cls, expr):
+        [domain, [element]] = construct_domain([expr])
+        return cls.new(element, domain)
+
+    def to_domain(self, domain):
+        element = domain.convert_from(self.element, self.domain)
+        return self.new(element, domain)
+
+    def unify(self, other):
+        domain = self.domain.unify(other.domain)
+        return self.to_domain(domain), other.to_domain(domain)
+
+    def __add__(self, other):
+        if not isinstance(other, DomainScalar):
+            return NotImplemented
+        self, other = self.unify(other)
+        return self.new(self.element + other.element, self.domain)
+
+    def __sub__(self, other):
+        if not isinstance(other, DomainScalar):
+            return NotImplemented
+        self, other = self.unify(other)
+        return self.new(self.element - other.element, self.domain)
+
+    def __mul__(self, other):
+        from .domainmatrix import DomainMatrix # it raises circular imports if defined at top or bottom
```

Comment:
We can not have an import in a method like this because importing a module is slow. If there is a circular import problem then maybe some other imports need to be moved.